azure-pipelines.yml

```diff
@@ -70,13 +70,13 @@ jobs:
     name: 'pool-ubuntu-2004'
   steps:
   - task: UsePythonVersion@0
-    displayName: 'Use Python 3.11'
+    displayName: 'Use Python 3.12'
     inputs:
-      versionSpec: 3.11
+      versionSpec: 3.12
   - bash: |
       #!/usr/bin/env bash
       set -ev
-      pip install wheel==0.30.0 requests packaging
+      pip install wheel==0.30.0 requests packaging setuptools
```

Comment:
wheel 0.30.0 is not compatible with Python 3.12. Install `setuptools` as a workaround.
```
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/./scripts/ci/test_index.py", line 23, in <module>
    from wheel.install import WHEEL_INFO_RE
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/wheel/install.py", line 17, in <module>
    from .paths import get_install_paths
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/wheel/paths.py", line 7, in <module>
    import distutils.command.install as install
ModuleNotFoundError: No module named 'distutils'
```

Reply:
Why installing `setuptools` can work around it?

Reply:
`setuptools` provides a "vendored" `distutils`.
```
root@7a5d336f2f35:/# python -c "import distutils;print(distutils.__path__)"
['/usr/local/lib/python3.12/site-packages/setuptools/_distutils']
```